x/fungible/keeper/msg_server_update_system_contract.go

```diff
@@ -77,5 +76,6 @@ func (k Keeper) UpdateSystemContract(goCtx context.Context, msg *types.MsgUpdate
 		k.Logger(ctx).Error("failed to emit event", "error", err.Error())
 		return nil, sdkerrors.Wrapf(types.ErrEmitEvent, "failed to emit event (%s)", err.Error())
 	}
+	commit()
```

Comment:
If commit after successfully emitting event is more correct, I‘m wondering if we have other places doing the same early commit.

Reply:
go hunt them down!